frontend/lib/identification/card_detail_view.dart

```diff
@@ -1,55 +1,120 @@
 import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
 import 'package:flutter_svg/svg.dart';
 import 'package:intl/intl.dart';
+import 'package:otp/otp.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
 
+import '../verification/verification_card_details.dart';
+import '../verification/verification_encoder.dart';
 import 'card_details.dart';
+import 'card_details_model.dart';
 import 'id_card.dart';
+import 'otp_generator.dart';
 
 class CardDetailView extends StatelessWidget {
   final CardDetails cardDetails;
   final VoidCallback onOpenQrScanner;
+  final OTPGenerator _otpGenerator;
 
-  const CardDetailView({Key key, this.cardDetails, this.onOpenQrScanner})
-      : super(key: key);
+  CardDetailView({Key key, this.cardDetails, this.onOpenQrScanner})
+      : _otpGenerator = OTPGenerator(
+            cardDetails.base32TotpSecret, 60 * 5, 10, Algorithm.SHA256),
```

Comment:
What are these magic numbers?

Reply:
Its the reset interval of the qr code and the opt length. It's a static constant after the changes in pull request #224.